xbmc/URL.h

```diff
@@ -204,4 +215,8 @@ class CURL
   std::string m_strProtocolOptions;
   CUrlOptions m_options;
   CUrlOptions m_protocolOptions;
+
+  //Response data
+  std::map<std::string, std::string> m_responseValues;
+  int m_responseCode = -1;
```

Comment:
Magic numbers are evil. I would introduce a constant with a speaking name for "invalid response code".